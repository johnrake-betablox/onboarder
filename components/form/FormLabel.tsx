interface Props {
  field: string;
  text: string;
  required?: boolean;
}
export default function FormLabel({ field, text, required = false }: Props) {
  return (
    <label
      htmlFor={field}
      className="block text-xs font-bold uppercase text-gray-600"
    >
      {text} {required && <span className="text-red-600">*</span>}
    </label>
  );
}
