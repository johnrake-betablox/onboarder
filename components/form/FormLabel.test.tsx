import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import FormLabel from "./FormLabel";

test("renders a snapshot", () => {
  const { asFragment, getByText } = render(
    <FormLabel field="email" text="Email Address" />,
  );
  expect(getByText("Email Address")).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
