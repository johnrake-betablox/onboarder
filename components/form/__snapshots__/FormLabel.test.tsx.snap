// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<DocumentFragment>
  <label
    class="block text-xs font-bold uppercase text-gray-600"
    for="email"
  >
    Email Address 
  </label>
</DocumentFragment>
`;
